src/utils/editor/source-search.js

```diff
@@ -94,7 +94,7 @@ function searchOverlay(query, modifiers) {
         matchLength = len;
         return "highlight highlight-start";
       }
-      while (!stream.match(query, false) && stream.peek()) {
+      while (!stream.match(regexQuery, false) && stream.peek()) {
```

Comment:
I've spent entirely too long looking for this, but this was the fix for whole word matching... 😆